Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java

```diff
@@ -707,39 +706,41 @@ public Process convertDtoToBean(ProcessDTO processDTO) throws DAOException {
+        JsonObject processJSONObject = jsonObject.getJsonObject("_source");
+        processDTO.setTitle(processJSONObject.getString("title"));
+        processDTO.setOutputName(processJSONObject.getString("outputName"));
+        processDTO.setWikiField(processJSONObject.getString("wikiField"));
+        processDTO.setCreationDate(processJSONObject.getString("creationDate"));
         processDTO.setPropertiesSize(getSizeOfRelatedPropertyForDTO(processJSONObject, "properties"));
         processDTO.setProperties(
                 convertRelatedJSONObjectToDTO(processJSONObject, "properties", serviceManager.getPropertyService()));
         processDTO.setSortedCorrectionSolutionMessages(getSortedCorrectionSolutionMessages(processDTO));
-        processDTO.setSortHelperArticles(getIntegerPropertyForDTO(processJSONObject, "sortHelperArticles"));
-        processDTO.setSortHelperDocstructs(getIntegerPropertyForDTO(processJSONObject, "sortHelperDocstructs"));
-        processDTO.setSortHelperImages(getIntegerPropertyForDTO(processJSONObject, "sortHelperImages"));
-        processDTO.setSortHelperMetadata(getIntegerPropertyForDTO(processJSONObject, "sortHelperMetadata"));
+        processDTO.setSortHelperArticles(processJSONObject.getInt("sortHelperArticles"));
+        processDTO.setSortHelperDocstructs(processJSONObject.getInt("sortHelperDocstructs"));
+        processDTO.setSortHelperImages(processJSONObject.getInt("sortHelperImages"));
+        processDTO.setSortHelperMetadata(processJSONObject.getInt("sortHelperMetadata"));
         processDTO.setTifDirectoryExists(
                 checkIfTifDirectoryExists(processDTO.getId(), processDTO.getTitle(), processDTO.getProcessBaseUri()));
         if (!related) {
             processDTO = convertRelatedJSONObjects(processJSONObject, processDTO);
         } else {
             ProjectDTO projectDTO = new ProjectDTO();
-            projectDTO.setId(getIntegerPropertyForDTO(processJSONObject, "project.id"));
-            projectDTO.setTitle(getStringPropertyForDTO(processJSONObject, "project.title"));
-            projectDTO.setActive(getBooleanPropertyForDTO(processJSONObject, "project.active"));
+            projectDTO.setId(processJSONObject.getInt("project.id"));
+            projectDTO.setTitle(processJSONObject.getString("project.title"));
+            projectDTO.setActive(processJSONObject.getBoolean("project.active"));
             processDTO.setProject(projectDTO);
         }
         return processDTO;
     }
 
-    private ProcessDTO convertRelatedJSONObjects(JSONObject jsonObject, ProcessDTO processDTO) throws DataException {
-        Integer project = getIntegerPropertyForDTO(jsonObject, "project.id");
-        processDTO.setProject(serviceManager.getProjectService().findById(project));
+    private ProcessDTO convertRelatedJSONObjects(JsonObject jsonObject, ProcessDTO processDTO) throws DataException {
+        Integer project = jsonObject.getInt("project.id");
+        if (project > 0) {
```

Comment:
Should here not a check if value of property `project` is not `null` as comparing `null` with `0` can cause trouble?

Reply:
```
Integer projectId = process.getProject() != null ? process.getProject().getId() : 0;
```
Here it doesn't store anymore nulls - 0 instead.

Reply:
I hope, you are right as `Integer` can be `null`.